### PR TITLE
auto OCP-30832 and OCP-27986

### PR DESCRIPTION
**Pull request by jianlinliu**
1. Some code change to cover OCP-30832.
2. The default behavior is the same as OCP-27986, so just place a marker to filter the coverage easily.
3. Per [discussion](https://github.com/openshift/release/pull/45597#discussion_r1392064890), added "-v" option to curl command, also unset https_proxy to avoid proxy noise.
4. Use `run_command` wrapper to print out upgrade command line.
5. Added a pre-check to ensure OCP-30832 and OCP-27986 is run with a signed image.
6. Make `check_signed` function work with tagname pullspec.
7. Added retry logic in `check_signed` function

### CODE REVIEW

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-arm64-stable-4.12-upgrade-from-stable-4.11-aws-ipi-disconnected-sts-ep-f28

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-stable-4.14-upgrade-from-stable-4.13-azure-ipi-disconnected-fullyprivate-f28

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-arm64-stable-4.12-upgrade-from-stable-4.11-aws-ipi-disconnected-sts-ep-f28

**Comment by jianlinliu** [comment]
cc @jiajliu @shellyyang1989 @JianLi-RH to review.

File: ci-operator/step-registry/cucushift/upgrade/mirror-images/by-oc-adm/cucushift-upgrade-mirror-images-by-oc-adm-commands.sh [hunk 1]
```diff
@@ -151,9 +152,22 @@ do
 
     extract_oc
 
-    mirror_image
+    if [[ "${ENABLE_OTA_TEST}" =~ "OCP-30832" ]]; then
+        MIRROR_RELEASE_IMAGE_REPO="${MIRROR_REGISTRY_HOST}/ota_auto/ocp"
+        mirror_apply_sig_flag="true"
```
Comment (jiajliu) [inline ci-operator/step-registry/cucushift/upgrade/mirror-images/by-oc-adm/cucushift-upgrade-mirror-images-by-oc-adm-commands.sh:157]: how about rename it to `mirror_apply_sig_together`?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-stable-4.12-upgrade-from-stable-4.11.yaml [hunk 2]
```diff
@@ -106,6 +107,7 @@ tests:
     cluster_profile: azure-qe
     env:
       BASE_DOMAIN: qe.azure.devcluster.openshift.com
+      ENABLE_OTA_TEST: OCP-27986
     test:
     - chain: openshift-upgrade-qe-sanity-disconnected
```
Comment (jiajliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-stable-4.12-upgrade-from-stable-4.11.yaml:112]: Should we add a skip condition if the target version is not signed? From [log](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/logs/periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-stable-4.13-upgrade-from-stable-4.12-azure-ipi-fullyprivate-proxy-f28/1723769588191072256), it seems not signed always even is stable-to-stable upgrade.
```
Seem like registry.build01.ci.openshift.org/ci-op-trwh6vvv/release@sha256:f323afe3e80d44da7d0caaed8c353fedef37e5fd5e7f765e8440be86f546cba0 is not signed
You're updating to an unsigned images, you must override the verification using --force flag
warning: The requested upgrade image is not one of the available updates.You have used --allow-explicit-upgrade for the update to proceed anyway
warning: --force overrides cluster verification of your supplied release image and waives any update precondition failures.
```

Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-stable-4.12-upgrade-from-stable-4.11.yaml:112]: Ideally, I think all the stable builds should be signed, so I even think once the OTA case is enable, I prefer to run some pre-check to ensure the payload is signed. That is also a requirement for such cases.

On my localhost, I run some pre-check, it shows me singed, I am not sure what happened in the job, later I will check more.
```
[root@preserve-jialiu ~]# TARGET=registry.build01.ci.openshift.org/ci-op-trwh6vvv/release@sha256:f323afe3e80d44da7d0caaed8c353fedef37e5fd5e7f765e8440be86f546cba0
[root@preserve-jialiu ~]# check_signed
registry.build01.ci.openshift.org/ci-op-trwh6vvv/release@sha256:f323afe3e80d44da7d0caaed8c353fedef37e5fd5e7f765e8440be86f546cba0 is signed
```

Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-stable-4.12-upgrade-from-stable-4.11.yaml:112]: > From [log](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/logs/periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-stable-4.13-upgrade-from-stable-4.12-azure-ipi-fullyprivate-proxy-f28/1723769588191072256), it seems not signed always even is stable-to-stable upgrade.

I checked another [log](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/logs/periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-stable-4.13-upgrade-from-stable-4.12-azure-ipi-fullyprivate-proxy-f28/1701663851268804608/artifacts/azure-ipi-fullyprivate-proxy-f28/cucushift-upgrade-toimage/build-log.txt) , sounds like a flake.

File: ci-operator/step-registry/cucushift/upgrade/mirror-images/by-oc-adm/cucushift-upgrade-mirror-images-by-oc-adm-commands.sh (hunk 1 above)
Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/upgrade/mirror-images/by-oc-adm/cucushift-upgrade-mirror-images-by-oc-adm-commands.sh]: updated.

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-arm64-stable-4.12-upgrade-from-stable-4.11-aws-ipi-disconnected-sts-ep-f28

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-stable-4.14-upgrade-from-stable-4.13-azure-ipi-disconnected-fullyprivate-f28

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-arm64-stable-4.12-upgrade-from-stable-4.11-aws-ipi-disconnected-sts-ep-f28

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-stable-4.14-upgrade-from-stable-4.13-azure-ipi-disconnected-fullyprivate-f28

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-stable-4.12-upgrade-from-stable-4.11.yaml (hunk 2 above)
Comment (jiajliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-stable-4.12-upgrade-from-stable-4.11.yaml:112]: It seems the job triggered before the payload signed. 
```
https://mirror.openshift.com/pub/openshift-v4/signatures/openshift/release/sha256%3Df323afe3e80d44da7d0caaed8c353fedef37e5fd5e7f765e8440be86f546cba0/
Nov 13 09:03:05 2023
```

File: ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh [hunk 3]
```diff
@@ -614,6 +622,12 @@ do
     if ! check_signed; then
         echo "You're updating to an unsigned images, you must override the verification using --force flag"
         FORCE_UPDATE="true"
+        if [[ "${ENABLE_OTA_TEST}" =~ "OCP-30832" ]] || [[ "${ENABLE_OTA_TEST}" =~ "OCP-27986" ]]; then
+            echo "ENABLE_OTA_TEST is enabled on this job, these cases need to run against a signed target image!"
+            exit 1
```
Comment (jiajliu) [inline ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh:627]: exit 1 intentionally? do u want  the upgrade job with the two cases enabled to be dedicated for the cases?

Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh:627]: Yes, I did that intentionally, ideally we can add more cases for the checking, when we are going to add more cases to ensure upgrade without --force option, we can extend the list later.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-stable-4.12-upgrade-from-stable-4.11.yaml (hunk 2 above)
Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-stable-4.12-upgrade-from-stable-4.11.yaml:112]: I added a retries logic to relief such edge cases.

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-4.14-upgrade-from-stable-4.14-azure-ipi-fullyprivate-proxy-f28

File: ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh (hunk 3 above)
Comment (jiajliu) [reply ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh:635]: ack. it seems the same logic in [step](https://github.com/openshift/release/blob/610b94a984ac635f8dac65ff36a9ffad14de020c/ci-operator/step-registry/cucushift/upgrade/mirror-images/by-oc-adm/cucushift-upgrade-mirror-images-by-oc-adm-commands.sh#L156)  which is located before this step. so the job will exit in advance, and this exit will not be executed, right?

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-arm64-stable-4.12-upgrade-from-stable-4.11-aws-ipi-disconnected-sts-ep-f28

File: ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh (hunk 3 above)
Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh:635]: Yeah, they are the same code, but the logic is applied to different case:
`cucushift-upgrade-mirror-images-by-oc-adm` ensure the mirroring payload should be signed, so that the following step signature configmap can be created successfully, or else, should exit in advance.
While `cucushift-upgade-toimage` step is ensure `oc adm upgrade` is run without `--force` option when the payload is signed, or else, exit immediately.

In the future, if we want to cover ipi behind proxy upgrade between stable builds, we just need to extend this list in `cucushift-upgade-toimage`, then we can easily cover all kinds of upgrade cases without --force option.

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-arm64-stable-4.12-upgrade-from-stable-4.11-aws-ipi-disconnected-sts-ep-f28

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-4.14-upgrade-from-stable-4.14-azure-ipi-fullyprivate-proxy-f28

**Comment by jianlinliu** [comment]
@jiajliu @shellyyang1989 @JianLi-RH The PR is ready to merge, pls help review.

**Comment by jianlinliu** [comment]
cc @liangxia to review
